futures-cpupool/src/lib.rs

```diff
@@ -258,8 +269,10 @@ impl<F: Future> Future for MySender<F, Result<F::Item, F::Error>> {
 
     fn poll(&mut self) -> Poll<(), ()> {
         if let Ok(Async::Ready(_)) = self.tx.as_mut().unwrap().poll_cancel() {
-            // Cancelled, bail out
-            return Ok(().into())
+            if !self.keep_running_flag.load(Ordering::SeqCst) {
```

Comment:
I think it's ok to switch the ordering of these two conditions, check `keep_running_flag` and then check `poll_cancel`. That way we can avoid messing with that piece if the flag is set.